tests/ludwig/automl/test_base_config.py

```diff
@@ -1,29 +1,45 @@
+import random
+
 import pytest
 
-from ludwig.automl.base_config import should_exclude, infer_type
+from ludwig.automl.base_config import infer_type, should_exclude
 from ludwig.automl.utils import FieldInfo
-from ludwig.constants import TEXT, NUMERICAL, CATEGORY, BINARY, IMAGE
+from ludwig.constants import BINARY, CATEGORY, IMAGE, NUMERICAL, TEXT
+from ludwig.data.dataset_synthesizer import generate_string
 
 ROW_COUNT = 100
 TARGET_NAME = 'target'
 
 
-@pytest.mark.parametrize("num_distinct_values,distinct_values,avg_words,img_values,missing_vals,expected", [
-    (ROW_COUNT, ['1.1','2.2'], 0, 0, 0.0, NUMERICAL),
-    (10, ['0','1','2','3','4','5','6','7','8','9'], 0, 0, 0.0, CATEGORY),
-    (2, ['0', '1'], 0, 0, 0.0, BINARY),
-    (2, ['human', 'bot'], 0, 0, 0.0, CATEGORY),
-    (2, ['1.5', '3.7'], 0, 0, 0.1, NUMERICAL),
-    (ROW_COUNT, [], 3, 0, 0.0, TEXT),
-    (ROW_COUNT, [], 1, ROW_COUNT, 0.0, IMAGE),
-])
-def test_infer_type(num_distinct_values, distinct_values, avg_words, img_values, missing_vals, expected):
+@pytest.mark.parametrize(
+    "num_distinct_values,distinct_values,img_values,missing_vals,expected",
+    [
+        # Random numbers.
+        (ROW_COUNT, [str(random.random()) for _ in range(ROW_COUNT)], 0, 0.0, NUMERICAL),
+        # Random numbers with NaNs.
+        (ROW_COUNT, [str(random.random()) for _ in range(ROW_COUNT - 1)] + ['NaN'], 0, 0.0,
+         NUMERICAL),
+        # Finite list of numbers.
+        (10, ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'], 0, 0.0, CATEGORY),
```

Comment:
Agree this should be category, but if the small set of numbers is not consecutive, I don't agree on category

Reply:
Discussed offline. Reverted!